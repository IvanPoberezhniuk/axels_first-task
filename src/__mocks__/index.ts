import { Comment, Image, ImageDetails } from '../redux/ducks/images';

const image: Image = {
  id: 222,
  url: '/fakeUri/images'
};

const comment: Comment = {
  id: 222,
  text: 'test',
  date: new Date().getSeconds()
};

const imagesDetails: ImageDetails = {
  id: 222,
  url: '/fakeUri/images',
  comments: [{
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }, {
    id: 222,
    text: 'test',
    date: new Date().getSeconds()
  }]
};
