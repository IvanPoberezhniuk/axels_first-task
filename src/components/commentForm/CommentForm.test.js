import { CommentForm } from '../index';
import { mockedImage } from '../../__mocks__';

const setUp = () => mount(<CommentForm />);
let component;
const mockUseDispatch = jest.fn();

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockUseDispatch,
  useSelector: () => ([mockedImage, mockedImage])
}));

beforeEach(() => {
  component = setUp();
});

describe('CommentForm component', () => {
  it('SNAPSHOT', () => {
    expect(component).toMatchSnapshot();
  });

  it('test', () => {
    const res = component.find('[className="btn btn-primary btn-block"]').simulate('click');
  });

  it('test', () => {
    const res = component.find('form.comment-form').simulate('submit');
  });
});
