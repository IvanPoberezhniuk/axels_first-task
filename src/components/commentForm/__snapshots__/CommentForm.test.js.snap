// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm component SNAPSHOT 1`] = `
<CommentForm>
  <Form
    className="comment-form"
    inline={false}
    noValidate={true}
    onSubmit={[Function]}
  >
    <form
      className="comment-form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <FormGroup
        controlId="name"
      >
        <div
          className="form-group"
        >
          <FormControl
            isInvalid={false}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Ваше имя"
            type="text"
            value=""
          >
            <input
              className="form-control"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Ваше имя"
              type="text"
              value=""
            />
          </FormControl>
          <Feedback
            type="invalid"
          >
            <div
              className="invalid-feedback"
            />
          </Feedback>
        </div>
      </FormGroup>
      <FormGroup
        controlId="comment"
      >
        <div
          className="form-group"
        >
          <FormControl
            isInvalid={false}
            name="text"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Ваше комментарий"
            type="text"
            value=""
          >
            <input
              className="form-control"
              id="comment"
              name="text"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Ваше комментарий"
              type="text"
              value=""
            />
          </FormControl>
          <Feedback
            type="invalid"
          >
            <div
              className="invalid-feedback"
            />
          </Feedback>
        </div>
      </FormGroup>
      <Button
        active={false}
        block={true}
        disabled={false}
        type="submit"
        variant="primary"
      >
        <button
          className="btn btn-primary btn-block"
          disabled={false}
          type="submit"
        >
          Оставить комментарий
        </button>
      </Button>
    </form>
  </Form>
</CommentForm>
`;
